Persephone/Controllers/QueueViewController.swift

```diff
@@ -81,25 +82,28 @@ class QueueViewController: NSViewController, NSOutlineViewDataSource, NSOutlineV
     guard let queue = notification.userInfo?[Notification.queueKey] as? [MPDClient.Song]
       else { return }
 
-    self.queue = queue
+    var newQueue: [SongItem] = []
 
+    for (index, mpdSong) in queue.enumerated() {
+      let songItem = SongItem(song: mpdSong, queuePos: index, isPlaying: index == queuePos)
```

Comment:
This is interesting, if I play a new album while a song is playing, what does MPD do with my current song & queue position? Is `queuePos` an instance variable? Will it always be valid with respect to the new queue?

Reply:
That's a good question. I haven't tried hitting play while an album is already playing.

Reply:
It seems to handle this just fine. I think clearing the queue resets the queuePos to -1.